### ISSUE TITLE
[Feature Request] Meta Annotation Support

### ISSUE THREAD

**Issue by F43nd1r** [ExpediaGroup/graphql-kotlin#1558]
**Is your feature request related to a problem? Please describe.**
I want to create a meta annotation like this 
```kotlin
@GraphQLIgnore
@Autowired
@Target(AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
annotation class GraphQLAutowired
```
because we plan to use autowiring into arguments quite a bit. However, currently meta annotations for @GraphQLIgnore are not supported.

**Describe the solution you'd like**
I'd like to see this library support meta annotations out of the box. 

**Describe alternatives you've considered**
If that is not possible, instead it would also help if there is a hook for `getValidArguments`.

**Comment by samuelAndalon** [comment]
Hello, have you just using  the combination of
```kotlin
@GraphQLIgnore
@Autowired
```
in your resolvers ? i guess the goal here is to be less verbose ?

**Comment by F43nd1r** [comment]
Yes, exactly. Meta annotations are pretty standard in spring for reducing code duplication, so I was a bit surprised to see it not working here.

I understand spring support is just an addon but maybe it's still possible to support it? Or provide the hook so we can do it ourselves?

**Comment by samuelAndalon** [comment]
sounds good, will work on adding the annotation.

**Comment by F43nd1r** [comment]
Any progress on this?

**Comment by autaut03** [comment]
@samuelAndalon Would it be fine if I added a hook `isValidParameter` that behaves identically to `isValidFunction` and `isValidProperty` to allow to auto-ignore framework provided annotations **without** resorting to custom meta-annotations? I'll make a PR if that's something you're willing to support :)
